File: languages/c++/patterns/3 State/StateAppend2.cpp
#include <iostream>
using namespace std;

#include "StateAppend1.h"
#include "StateAppend2.h"
#include "Context.h"

void StateAppend2::write(Context* c, string s){
	cout << s << "-2\r\n";
	//change state
	c->setState(new StateAppend1);
}

File: languages/java/3 State/Main/src/main/StateAppend2.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main;

/**
 *
 * @author lupin
 */
public class StateAppend2 implements State {
    public void write(Context c, String s){
        System.out.println(s + "-2\r\n");
	//change state
	c.setState(new StateAppend1());        
    }     
}
